client/state/ui/post-type-list/test/reducer.js

```diff
@@ -81,5 +87,59 @@ describe( 'reducer', () => {
 
 			expect( state.activeSharePanels ).to.eql( [ postGlobalId ] );
 		} );
+
+		test( 'should add postGlobalId to the state when toggling selectedPosts with a new globalId', () => {
+			const existingPostGlobalId = 'abcdef0123456789';
+			const postGlobalId = '0123456789abcdef';
+			const state = postTypeList(
+				{ selectedPosts: [ existingPostGlobalId ] },
+				{
+					type: POST_TYPE_LIST_SELECTION_TOGGLE,
+					postGlobalId: postGlobalId,
+				}
+			);
+
+			expect( state.selectedPosts ).to.eql( [ existingPostGlobalId, postGlobalId ] );
+		} );
+
+		test( 'should remove postGlobalId from the state when toggling selectedPosts with an existing globalId', () => {
+			const existingPostGlobalId = 'abcdef0123456789';
+			const postGlobalId = '0123456789abcdef';
+			const state = postTypeList(
+				{ selectedPosts: [ existingPostGlobalId, postGlobalId ] },
+				{
+					type: POST_TYPE_LIST_SELECTION_TOGGLE,
+					postGlobalId: postGlobalId,
+				}
+			);
+
+			expect( state.selectedPosts ).to.eql( [ existingPostGlobalId ] );
+		} );
+
+		test( 'should clear all selectedPosts state when toggling on selection mode', () => {
+			const postGlobalId = '0123456789abcdef';
+			const state = postTypeList(
+				{ selectedPosts: [ postGlobalId ], isMultiSelectEnabled: false },
+				{
+					type: POST_TYPE_LIST_MULTI_SELECTION_MODE_TOGGLE,
+				}
+			);
+
+			expect( state.selectedPosts ).to.be.empty;
+			expect( state.isMultiSelectEnabled ).to.be.true;
+		} );
+
+		test( 'should maintain all selectedPosts state when toggling off selection mode', () => {
```

Comment:
This is not the behavior I would expect.  When we toggle post selection to "off", shouldn't we clear the list of post IDs?

Alternatively, we could just always preserve this list of post IDs.  This would lead to simpler and more obvious reducer code, and might be a better user experience - arguably, simply clicking the Select button twice in a row should not reset the list of selected IDs.